packages/@magic-sdk/provider/src/modules/auth.ts

```diff
@@ -15,8 +15,9 @@ export class AuthModule extends BaseModule {
    * this method will return a Decentralized ID token (with a default lifespan
    * of 15 minutes).
    */
-  public loginWithMagicLink(configuration: LoginWithMagicLinkConfiguration) {
+  public async loginWithMagicLink(configuration: LoginWithMagicLinkConfiguration) {
```

Comment:
Will have to revert this as it will break the `PromiEvent` APIs

Reply:
TODO: Add a test case to verify the result of `loginWithMagicLink` is a `PromiEvent`.